src/mutate-errors-to-todos.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  TodoData,
+  todoDirFor,
+  todoFileNameFor,
+  _buildTodoDatum,
+} from '@ember-template-lint/todo-utils';
+import { ESLint } from 'eslint';
+
+/**
+ * Mutates all errors present in the todo dir to todos in the results array.
+ * @param results ESLint results array
+ */
+export async function mutateTodoErrorsToTodos(
+  results: ESLint.LintResult[],
+  todoMap: Map<string, TodoData>
+) {
+  results.forEach((result) => {
+    result.messages.forEach((message) => {
+      if (message.severity !== 2) {
+        return;
+      }
+
+      // we only mutate errors that are present in the todo map, so check if it's there first
+      const todoDatum = _buildTodoDatum(result, message);
+      const todoDirHash = todoDirFor(result.filePath);
+      const todoFileHash = todoFileNameFor(todoDatum);
+      const todoHash = `${todoDirHash}/${todoFileHash}`;
+
+      if (!todoMap.get(todoHash)) {
+        return;
+      }
+
+      (message.severity as any) = -1;
```

Comment:
Why do you need to cast this to `any`? Isn't this why we added the extended type?

Reply:
The extended interface served to introduce a new key: `todoCount`. I can't do the same with the `type Severity` in TS. See my comment above. I did change this to `number` though to make it consistent with other places.

Reply:
I think it would be better to cast message at the top to the widened type we defined.

Reply:
Ok I agree. I've modified the code to cast to the modified message.